### PR TITLE
render markdown, strip html tags in indexing

### LINKED ISSUE #917: Search results are overflowing.
If you use mobile emulation on devtools and search for something, the results will overflow and look cropped. This means the preview images are hidden.

image

### LINKED ISSUE #924: Search results show raw content of {% Img %} shortcode
Search results show raw content of {% Img %} shortcode - seems like algolia is getting raw content instead of already rendered html.

### PR DESCRIPTION
**Pull request by samthor**
Fixes #917 
Fixes #924 

I know @devnook tried to solve this in #923. I disagree with @robdodson 's comment, I think we should `break-word` just in case.

Fixes a minor display issue too; if we don't have content on page (which happens for some of the structural pages), fall back to the meta description.

### CODE REVIEW

**Comment by robdodson** [comment]
> I know @devnook tried to solve this in #923. I disagree with @robdodson 's comment, I think we should break-word just in case.

What case are you worried about? One that I could think of is if someone tries to have a `reallyLongMethodName` but I think our site's `code` style may already cause that to break (I _think_ ... would be good to test).

**Comment by robdodson** [comment]
Visual comparison with and without break-all.

With break-all pretty much every word near the right-hand edge will get broken in a weird way.

Screen Shot 2021-08-12 at 4 11 29 PM

If you remove the style then the words wrap naturally

Screen Shot 2021-08-12 at 4 12 55 PM

**Comment by samthor** [comment]
You're demoing `break-all`? The `break-word` option, despite its confusing name, avoids breaking words where possible. It only breaks words too long to render at once. The [example on MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break) is good for this: with `break-word`, the first two words remain on their own lines, it's only the third really huge word that's split up.

**Comment by robdodson** [comment]
ah sorry, I didn't notice that you said `break-word`. I thought you were saying you disagreed with my comment not to use `break-all` in Ewa's PR. Yeah, `break-word` is fine.

File: site/_utils/drafts.js [hunk 1]
```diff
@@ -5,12 +5,11 @@
...
-  } else {
+const filterDrafts = item => {
+  if (process.env.NODE_ENV === 'production') {
     return !item.data.draft;
   }
+  return true; // include everything in non-prod
 };
 
-module.exports = {drafts};
+module.exports = {filterDrafts};
```
Comment (devnook) [inline site/_utils/drafts.js:15]: I'd suggest filterOutDrafts or removeDrafts. Also, this looks like an eleventy filter, should it be in site/_filters?  How does this compare to _filters/removeDrafts?

File: site/_collections/algolia.js [hunk 2]
```diff
@@ -47,38 +51,55 @@ function limitText(content, limit = 7500) {
...
       objectID: createHash('md5').update(item.url).digest('hex'),
     };
 
+    // The item is dumped to JSON, but we can't get at the underlying post's
+    // templateContent until the rendering state of 11ty. Define a getter
+    // which will be called at the later point.
+    Object.defineProperty(algoliaCollectionItem, 'content', {
+      get() {
+        // Strip HTML tags and limit to a sensible size for indexing.
+        const text = striptags(item.templateContent ?? '');
```
Comment (devnook) [inline site/_collections/algolia.js:93]: Is this already rendered content, e.g. with shortcodes rendered in?

**Comment by samthor** [comment]
yes - templateContent is itself a getter which renders the final page
content but only when 11ty is rendering pages. If we get it here, we crash.

File: site/_utils/drafts.js (hunk 1 above)
Comment (samthor) [reply site/_utils/drafts.js:15]: It's not really an 11ty filter, which is confusing. It's just a helper for collection filtering. I've made that clearer, and I like the updated name—thanks.

File: site/_collections/algolia.js (hunk 2 above)
Comment (samthor) [reply site/_collections/algolia.js:93]: yes - templateContent is itself a getter which renders the final page content but only when 11ty is rendering pages. If we get it here, we crash.

**Comment by robdodson** [comment]
@samthor any idea why the linter is failing? it seems to be missing a module

**Comment by samthor** [comment]
I expanded the typechecking a tiny bit. I've fixed the other types issues in an extra commit here.

**Review by devnook — APPROVED** [review]

**Pull request merged**